src/app/shared/entry-actions/entry-actions.service.ts

```diff
@@ -184,4 +189,41 @@ export class EntryActionsService {
       );
     }
   }
+
+  archiveEntry(entry: Entry) {
+    this.dialog.open(ArchiveEntryDialogComponent, {
+      width: bootstrap4largeModalSize,
+      data: {
+        entry: entry,
+        callback: (archived: Entry) => {
```

Comment:
I don't think we do callbacks with any other dialogs. I'd lean towards doing it the same way for consistency's sake, even though it's arguably more of a hassle to do the subscription.